### ISSUE TITLE
set_fact module breaks for Cisco n/w switches after upgrading ansible-core from 2.14.0 to 2.14.12

### ISSUE THREAD

**Issue by shaaga** [ansible/awx#14780]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx/issues) for duplicates.
- [X] I understand that AWX is open source software provided for free and that I might not receive a timely response.
- [X] I am **NOT** reporting a (potential) security vulnerability. (These should be emailed to `security@ansible.com` instead.)

### Bug Summary

We have AWX v17.1.0 running and recently to fix a security item, we upgraded ansible-core from **2.14.0** to **2.14.12** for the ansible venv present in AWX containers at path /var/lib/awx/venv/ansible. Post upgrade, existing job templates which connects to Cisco switches started failing with below error.

We use **v2.7.1 of ansible collection cisco_nxos**

Ansible job template:

```
- hosts: "elf"
  gather_facts: no
  connection: network_cli
  vars:
    ansible_user: "{{ network_user }}"
    ansible_password: "{{ network_password }}"
    ansible_become_password: "{{ network_password }}"
    ansible_command_timeout: 600
    ansible_connect_timeout: 600
    operation: "{{ action }}"
    tenant_type: "{{ 'stretched-cluster' if features.get('stretched_cluster') else 'traditional' }}"
  tasks:
    - name: Setting releases as fact
      set_fact:
        releases: "{{ releases }}"
```

Error:

```
{
  "msg": "Unexpected failure during module execution: maximum recursion depth exceeded while getting the str of an object",
  "exception": "Traceback (most recent call last):\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/executor/task_executor.py\", line 158, in run\n    res = self._execute()\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/executor/task_executor.py\", line 580, in _execute\n    socket_path = start_connection(self._play_context, options, self._task._uuid)\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/executor/task_executor.py\", line 1203, in start_connection\n    write_to_file_descriptor(master, options)\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/module_utils/connection.py\", line 58, in write_to_file_descriptor\n    src = cPickle.dumps(obj, protocol=0)\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/copyreg.py\", line 71, in _reduce_ex\n    state = base(self)\nRecursionError: maximum recursion depth exceeded while getting the str of an object\n",
  "stdout": "",
  "_ansible_no_log": false
}
```

### AWX version

17.1.0

### Select the relevant components

- [ ] UI
- [ ] UI (tech preview)
- [X] API
- [ ] Docs
- [X] Collection
- [X] CLI
- [ ] Other

### Installation method

docker development environment

### Modifications

no

### Ansible version

7.0.0

### Operating system

Ubuntu 22.04

### Web browser

Chrome

### Steps to reproduce

Upgrade ansible-core for "/var/lib/awx/venv/ansible" virtual env in awx container from 2.14.0 to 2.14.12 and use cisco_nxos v2.7.1 and write a job template like above

### Expected results

Expected results are normal:

![image](https://github.com/ansible/awx/assets/67102357/a40feb27-d8a1-46ee-a03d-96bbf023d5d2)

### Actual results

Play is failing with below error:

![image](https://github.com/ansible/awx/assets/67102357/a058faa1-a311-4197-a7ce-edae6490b58c)

Complete error trace:

{
  "msg": "Unexpected failure during module execution: maximum recursion depth exceeded while getting the str of an object",
  "exception": "Traceback (most recent call last):\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/executor/task_executor.py\", line 158, in run\n    res = self._execute()\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/executor/task_executor.py\", line 580, in _execute\n    socket_path = start_connection(self._play_context, options, self._task._uuid)\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/executor/task_executor.py\", line 1203, in start_connection\n    write_to_file_descriptor(master, options)\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/site-packages/ansible/module_utils/connection.py\", line 58, in write_to_file_descriptor\n    src = cPickle.dumps(obj, protocol=0)\n  File \"/var/lib/awx/venv/ansible/lib/python3.9/copyreg.py\", line 71, in _reduce_ex\n    state = base(self)\nRecursionError: maximum recursion depth exceeded while getting the str of an object\n",
  "stdout": "",
  "_ansible_no_log": false
}

**Comment by shaaga** [comment]
The issue seems similar to what is mentioned in the following issue but the fix suggests downgradng execution env in AWX but my AWX version is 17.1.0 and it has no concept of execution env

https://github.com/ansible/awx/issues/14689 @gundalow , @Dustin-Wi @ZachHoiberg

**Comment by shaaga** [comment]
The issue got fixed by upgrading ansible from 7.0.0 to 8.5.0
